### DISCUSSION TITLE
支付回调通知错误提示 valid request body.

### DISCUSSION THREAD

**Discussion by boyyangzz** [w7corp/easywechat#2681] [Q&A]
![image](https://user-images.githubusercontent.com/18475834/232945827-48ce6dd3-9759-49ba-9c1c-2da2e680415c.png)
源文件路径easywechat\src\Pay\Server.php
支付成功后，使用回调通知接口无法获取消息body内的json
$server = $app->getServer();
$server->handlePaid(function (Message $message, \Closure $next) {
   print_r($message);
});

**Comment by TheNorthMemory** [comment]
微信支付APIv2通知的body是XML，APIv3是JSON，探测请求头例如:

```php
[$mime] = $request->hasHeader('content-type') ? $request->getHeader('content-type') : ['noop'];

if (stripos($mime, 'xml') > 0) {
    //do something
}
if (stripos($mime, 'json') > 0) {
    //do something
}
if (stripos($mime, 'noop') > 0) {
    // do something
}
```
